### PR TITLE
Paginate test results page

### PR DESCRIPTION
**Pull request by nickscialli-usds**
## Related Issue or Background Info

- Backend: #516
- Frontend: #554

This was initially going to only be the backend work until I realized the FE work was necessary to ship alongside it.

## Changes Proposed

- Add `Pagination` component to `TestResultsList.tsx` frontend and update routing accordingly
- Modifies graphql query `testResults` and adds `testResultsCounts` as follows:

```
testResults(facilityId: ID!, pageNumber: Int, pageSize: Int): [TestResult]
testResultsCount(facilityId: ID!): Int
```

- Pulls those changes through the resolver, service, and repository
- Adds some backend tests and modifies existing frontend tests

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/db/repository/TestEventRepository.java [hunk 1]
```diff
@@ -8,49 +8,48 @@
...
+    @Query(value = "WITH FILTEREDEVENTS AS (" + " SELECT DISTINCT ON (test_order_id) * "
+            + " FROM {h-schema}test_event te " + " WHERE te.facility_id = :facilityId "
+            + " ORDER BY test_order_id, te.created_at desc" + ") " + " SELECT * FROM FILTEREDEVENTS "
+            // moving this filter into the CTE makes this query significantly
+            // more efficient (like 75% faster in one case), but then when we
+            // make it more complicated somebody will probably break it
+            + " ORDER BY created_at DESC ", countQuery = "SELECT count(*) FROM FILTEREDEVENTS", nativeQuery = true)
+    public List<TestEvent> getTestEventResults(UUID facilityId, Pageable pageable);
+
+    @Query(value = "WITH FILTEREDEVENTS AS (" + " SELECT DISTINCT ON (test_order_id) * "
```
Comment (nickscialli-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/repository/TestEventRepository.java:44]: Is this not optimal? Really i just want a count of the results specified in the query above

File: backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java [hunk 2]
```diff
@@ -42,14 +44,11 @@
...
-      DeviceTypeService dts,
-      TestOrderRepository repo,
-      PatientAnswersRepository parepo,
-      TestEventRepository terepo,
-      PersonService ps,
-      PatientLinkService pls,
+  public static final int DEFAULT_PAGINATION_PAGEOFFSET = 0;
+  public static final int DEFAULT_PAGINATION_PAGESIZE = 5000;
+
+  public TestOrderService(OrganizationService os, DeviceTypeService dts, TestOrderRepository repo,
```
Comment (nickscialli-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java:50]: My IDE is a bit aggressive in applying formatting on save, which I generally like, but unfortunately causes unnecessary diffs. If this proves too hard to review, I'm happy to try to back out any purely formatting changes.

File: backend/src/test/java/gov/cdc/usds/simplereport/service/TestOrderServiceTest.java [hunk 3]
```diff
@@ -39,6 +42,22 @@
   @Autowired private PersonService _personService;
   @Autowired private TestDataFactory _dataFactory;
 
+  private static final PersonName AMOS = new PersonName("Amos", null, "Quint", null);
```
Comment (TomNUSDS) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/TestOrderServiceTest.java:45]: More duplicate code. There is a factory class used by tests, but it doesn't need to be part of this PR, IMO.

**Review by TomNUSDS — APPROVED** [review]
LGTM - Only minor comments.

FYI. The naming convention we use for branches is `github-username/branch-name`

File: backend/src/test/java/gov/cdc/usds/simplereport/service/TestOrderServiceTest.java (hunk 3 above)
Comment (nickscialli-usds) [reply backend/src/test/java/gov/cdc/usds/simplereport/service/TestOrderServiceTest.java:45]: Do we have a java faker library on the backend? It'd be helpful to just create a number of random patients without caring about what their names are. Something like this for the factory class:

```java
public List<Person> createMinimalPeople(number: Int, Organization org, Facility fac) {
  // Faker magic here that generates some people with random names
}
```

**Pull request merged**